.devcontainer/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM node:12-bullseye-slim
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends \
+        locales build-essential git curl sudo python3-dev python3-venv python3-pip ca-certificates unzip openssh-client \
+        default-jdk libpq-dev postgresql postgresql-client graphviz libmagic-dev ruby-dev ruby-rubygems nginx-light redis python2 \
+        libjpeg-dev libtiff-dev libwebp-dev liblcms2-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8 \
+    && curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.16.deb \
+    && dpkg -i elasticsearch-5.6.16.deb \
+    && rm elasticsearch-5.6.16.deb \
+    && gem install compass \
+    && curl -o awscliv2.zip "https://awscli.amazonaws.com/awscli-exe-linux-$(uname -m)-2.0.30.zip" \
+    && unzip awscliv2.zip \
+    && ./aws/install \
+    && rm -rf aws awscliv2.zip \
+    && useradd -ms /bin/bash --groups sudo encoded \
+    && echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers \
+    && mkdir /app \
+    && chown encoded /app \
+    # Workaround for https://bugs.launchpad.net/ubuntu/+source/nginx/+bug/1899164
+    && chown -R encoded /var/log/nginx /var/lib/nginx
+
+WORKDIR /app
+USER encoded
+ENV LANG en_US.utf8
+ENV PATH=/app/venv/bin:/usr/share/elasticsearch/bin:/usr/lib/postgresql/13/bin:${PATH}
```

Comment:
Sorry I didn't notice last review, but we are on Postgres 11, I think we need to be explicit about the version when installing it from apt.

Reply:
Debian Bullseye on which this container is based ships with Postgres 13 so switching versions would require installing from other repositories. While possible, it doesn't seem worth the hassle for a development container given it works with this version. Same for running with Python 3.9 instead of 3.8.

Reply:
I'm just concerned about devs introducing code that uses backwards-incompatible features of newer PG/Python/whatever, which works in the dev environment but errors on demos/RC/prod. IMO the hassle of installing in the container is worth it to save developer effort trying to figure out why code is breaking elsewhere when it works locally.

Granted, almost nobody is touching the PG stuff, but for Python it is certainly easy to be tempted to use 3.9-only features. We do test against PG 11 and Python 3.8.5 in CircleCI, but test coverage alone is insufficient.